controller/src/main/java/org/jboss/as/controller/CapabilityReferenceRecorder.java

```diff
@@ -146,6 +145,16 @@ private void processCapabilityRequirement(OperationContext context, String attri
             }
         }
 
+        private RuntimeCapability<?> getDependentCapability(OperationContext context) {
+            ImmutableManagementResourceRegistration mrr = context.getResourceRegistration();
+            for (RuntimeCapability<?> capability : mrr.getCapabilities()) {
+                if (this.baseDependentName.equals(capability.getName())) {
+                    return capability;
+                }
+            }
+            throw ControllerLogger.ROOT_LOGGER.unknownCapability(this.baseDependentName);
```

Comment:
Should we pass the OperationContext address to the exception so that the users can identify which resource is causing the issue?

Reply:
We could - but unless the same capability is defined by multiple resources (which is rare), the problematic resource is usually obvious.